.circleci/config.yml

```diff
@@ -1,63 +1,91 @@
+      command-run:
+        type: string
     docker:
-      - image: python:3.6
+      - image: cimg/python:3.7
+        #auth:
+        #  username: $DOCKERHUB_USERNAME
+        #  password: $DOCKERHUB_PASSWORD
     steps:
       - checkout
       - run:
-          name: Build source distribution and install package from it
-          working_directory: ~/project/
-          # Ensure we can build a source distribution that can correctly install.
-          # "python setup.py sdist" creates dist/wikipron-x.y.z.tar.gz
-          command: |
-              pip install --progress-bar off --upgrade pip setuptools
-              python setup.py sdist
-              pip install dist/`ls dist/ | grep .tar.gz`
-      - run:
-          name: Install the full development requirements
-          working_directory: ~/project/
-          command: pip install --progress-bar off -r requirements.txt
-      - run:
-          name: Show installed Python packages
-          command: pip list
+          name: Create custom requirements
+          command: grep << parameters.command-name >> requirements.txt > << parameters.command-name >>_requirements.txt
+      - python/install-packages:
+          pkg-manager: pip
+          pip-dependency-file: << parameters.command-name >>_requirements.txt
+          cache-version: << parameters.command-name >>-v1
       - run:
-          name: Lint
           working_directory: ~/
-          # Avoid being able to import wikipron by relative import.
-          # Test code by importing the *installed* wikipron in site-packages.
-          command: flake8 project/setup.py project/wikipron project/tests
+          command: << parameters.command-run >>
+
+  build-python:
+    parameters:
+      python-version:
+        type: string
+    docker:
+      - image: cimg/python:<< parameters.python-version >>
+        #auth:
+        #  username: $DOCKERHUB_USERNAME
+        #  password: $DOCKERHUB_PASSWORD
+    steps:
+      - checkout
       - run:
-          name: Check styling consistency with black
-          working_directory: ~/
-          # Avoid being able to import wikipron by relative import.
-          # Test code by importing the *installed* wikipron in site-packages.
-          command: black --line-length=79 --check project/setup.py project/wikipron project/tests project/data
+          name: Create empty pyenv file
+          command: |
+              mkdir -p /home/circleci/.pyenv/versions/
+              touch /home/circleci/.pyenv/version
+      - python/install-packages:
+          pkg-manager: pip
+          pre-install-steps:
+              - run:
+                  name: Build source distribution and install package from it
+                  command: |
+                      pip install --progress-bar off --upgrade pip setuptools
+                      python setup.py sdist
+                      pip install dist/*.tar.gz
+      - run:
+          name: Show installed Python packages
+          command: pip3 list -v
```

Comment:
`pip` (used in the previous step) instead of `pip3` for consistency?

Reply:
I modified the command to `pip3` because in a latter image `pip` showed both Python 2 and 3 packages. Do you want me to remove also the `-v` flag that I used for debugging the package location? This is an [output sample](https://app.circleci.com/pipelines/github/kylebgorman/wikipron/558/workflows/c28a52d8-0b7c-40bf-95a7-6ce66168d0d5/jobs/1630/parallel-runs/0/steps/0-107)